### DISCUSSION TITLE
URLs to the most recent pub are unexpectedly permalinks to a historical revision

### DISCUSSION THREAD

**Discussion by metasj** [knowledgefutures/pubpub#1315] [Ideas]
When visiting the current version of a pub (which has the short link `http://<community>/pub/<title>` that is used in citations, social media posts, &c), browsers are redirected to the full revision-permalink to the latest revision (`http://<community>/pub/<title>/release/xxx`), instead.

As both reader and editor:  
- This can be confusing in situations where the browser address is used anywhere else.  Including some tools for autogenerating presentations or links from tabs.  

As a reader:
- It would be nice to have a one-click way to get a link to the current page (i.e.: the short-hand URL).  I have to follow the 'social' link and then cut it out of the whole.  
- While I expect both of the above URLs to redirect to the same page, it would be nice for the URL that appears in the browser to be the shorter one by default.  Unless perhaps the reader followed a permalink rather than the "latest revision" shorthand.

**Comment by gabestein** [comment]
Thanks for the thought! A lot of research went into the decision to build it as-is, but it's a good moment to provide some of the context behind our thinking, so I'm going to move it to a discussion to see what folks think.

Long story short: we think that this the best way to do versioning for the web in general, for scholarly publishing, and for reader UX, even if it bucks the current trend a little bit. 

For the web, it seems pretty clear-cut to us that each version of a document should have its own canonical URL. Masking that with a root URL for the "latest version" feels a little weird, even though it's common today. Forwarding like we do is exactly what 301s are for, and avoids some of the SEO and indexing issues folks face with out-of-date canonical urls, etc.

For scholarly publishing, it's even more clear that this behavior is desirable, because users need to be able to cite specific versions of documents. Masking makes that harder, and makes it harder to understand that there are versions to begin with. This approach of forwarding from a base url to a latest version is also quite common on preprint servers, making us feel like we're in pretty good company here.

From a UX perspective, even though it's a little outside the norm, we think it's better, for similar reasons to the scholarly use-case. If you link to a document on PubPub on social media, it's useful context to know which version you linked to. If that version becomes out-of-date, it's good for a reader to know that. On the backend, forwarding rather than masking allows us to easily show traffic to different versions.

We've put a lot of effort into making it clear, when readers land on an old version, what they're seeing, and providing a changelog that shows them updates to the document. In looking at the data, we find that the vast majority of users (90%+) follow the jump to the latest version just fine. It seems to have little effect on bounce rate.

On the admin side, our release success dialog provides both links and explains the difference and purpose between them.

I do think we could potentially improve the UX by finding a way to expose the two URLs to readers as part of the share dialog, which we've discussed before. But given the above, it hasn't yet felt super high priority.

As always, feel free to feed back on the above! We feel pretty good about the decision, but we're open to enhancements, for sure.

**Reply by amdeis** [reply]
RE scholarly publishing and including citation of a specific version, I wonder if the DOI or version-specific URL are being used more and which _should_ be used more in cases where more than one version exists. APA tends to advise DOI over most anything, but there's definitely value in version citing. Thank you for the clarification on URL! It helps both us on the editorial board of our journal and for us to explain to authors.
